src/Rules/CallableExistsCheck.php

```diff
@@ -0,0 +1,186 @@
+<?php declare(strict_types = 1);
+
+namespace PHPStan\Rules;
+
+use PhpParser\Node\Arg;
+use PhpParser\Node\Expr;
+use PhpParser\Node\Expr\Array_;
+use PhpParser\Node\Expr\BinaryOp\Concat;
+use PhpParser\Node\Expr\ConstFetch;
+use PhpParser\Node\Expr\PropertyFetch;
+use PhpParser\Node\Expr\StaticPropertyFetch;
+use PhpParser\Node\Expr\Variable;
+use PhpParser\Node\Name;
+use PhpParser\Node\Scalar\MagicConst\Class_;
+use PhpParser\Node\Scalar\String_;
+use PHPStan\Analyser\Scope;
+use PHPStan\Broker\Broker;
+use PHPStan\Type\ArrayType;
+use PHPStan\Type\MixedType;
+use PHPStan\Type\ObjectType;
+use PHPStan\Type\StringType;
+use PHPStan\Type\ThisType;
+use ReflectionClass;
+
+class CallableExistsCheck
+{
+
+	/**
+	 * @var \PHPStan\Broker\Broker
+	 */
+	private $broker;
+
+	public function __construct(Broker $broker)
+	{
+		$this->broker = $broker;
+	}
+
+	/**
+	 * @param \PhpParser\Node\Arg $argument
+	 * @param \PHPStan\Analyser\Scope $scope
+	 * @param string $messagePrefix
+	 * @return string[]
+	 */
+	public function checkCallableArgument(
+		Arg $argument,
+		Scope $scope,
+		string $messagePrefix
+	): array
+	{
+		$targetType = null;
+		$targetMethod = null;
+		$targetIsStaticMethod = false;
+		$targetIsFunction = false;
+
+		$callbackValue = $this->resolveValue($argument->value, $scope);
+
+		if (is_string($callbackValue)) {
+			$targetMethod = $callbackValue;
+			$targetIsFunction = true;
+		} elseif ($callbackValue instanceof MixedType || $callbackValue instanceof StringType) {
+			return [];
+		} elseif ($callbackValue instanceof ArrayType) {
+			if (!$argument->value instanceof Array_) {
+				return [];
+			}
+			$arrayItemsCount = count($argument->value->items);
+			if ($arrayItemsCount < 2) {
+				return [$messagePrefix . "array doesn't have required 2 items."];
+			}
+			if ($arrayItemsCount > 2) {
+				return [$messagePrefix . 'array has too many items, only two items expected.'];
+			}
+
+			$item1 = $this->resolveValue($argument->value->items[0]->value, $scope);
+			$item2 = $this->resolveValue($argument->value->items[1]->value, $scope);
+
+			if (is_string($item1)) {
+				$targetType = new ObjectType($item1);
+				$targetIsStaticMethod = true;
+			} elseif ($item1 instanceof ObjectType || $item1 instanceof ThisType) {
```

Comment:
I feel like this should be a different condition. What about not having the `else` branch and just asking the `Type` about `hasMethod`/`getMethod`?

Reply:
This will solve union/intersection types, StaticType...

Reply:
I'm not sure what you mean here. How do you think this conditionshould look like?

Reply:
Like this:

```
if (is_string($item1)) {
   ...
} elseif ($item1 instanceof MixedType || $item1 instanceof StringType) {
   ...
} else {
   // fallback for all other types - make sure to call $item1->hasMethod() + getMethod somewhere lower in this file
}
```